src/psyclone/transformations.py

```diff
@@ -120,9 +121,9 @@ def _validate(self, node_list):
 
         # Check that we aren't attempting to include any else/else if's without
         # their parent 'if'.
-        from psyclone.psyGen import IfClause
+        from psyclone.psyGen import IfBlock
         for node in node_list:
-            if isinstance(node, IfClause):
+            if isinstance(node, Schedule) and isinstance(node.parent, IfBlock):
```

Comment:
I know we discussed this and we have tests that check that it works but I don't understand it! Please could you add a comment explaining. Is it that, for the list of nodes to be valid, they must have a `Schedule` as their parent?

Reply:
I deleted this altogether, I think the original code was just trying to avoid transformations to IfClause but was allowing transformation to the body of the clause, now the IfClause doesn't exist anymore.

Reply:
Great.